packages/fetchai/protocols/fipa/dialogues.py

```diff
@@ -335,18 +327,66 @@ def get_dialogue(self, fipa_msg: Message, agent_addr: Address) -> Dialogue:
             other_initiated_dialogue = cast(
                 FipaDialogue, self.dialogues[other_initiated_dialogue_label]
             )
-            if other_initiated_dialogue.is_valid_next_message(fipa_msg):
-                result = other_initiated_dialogue
+            result = other_initiated_dialogue
         if self_initiated_dialogue_label in self.dialogues:
             self_initiated_dialogue = cast(
                 FipaDialogue, self.dialogues[self_initiated_dialogue_label]
             )
-            if self_initiated_dialogue.is_valid_next_message(fipa_msg):
-                result = self_initiated_dialogue
+            result = self_initiated_dialogue
         if result is None:
             raise ValueError("Should have found dialogue.")
         return result
 
+    def update(
+        self,
+        message: Message,
+        is_self_message: bool,
+        agent_addr: Address,
+        is_seller: bool,
+    ) -> Optional[Dialogue]:
+        """
+        Update the state of dialogues with a new message.
+
+        If the message is for a new dialogue, a new dialogue is created with 'message' as its first message and returned.
+        If the message is addressed to an existing dialogue, the dialogue is retrieved, extended with this message and returned.
+        If there are any errors, e.g. the message dialogue reference does not exists, the message is invalid w.r.t. the dialogue, return None.
+
+        :param message: a new message
+        :param agent_addr: the agent address
+        :return: the new or existing dialogue the message is intended for, or None in case of any errors.
+        """
+        fipa_msg = cast(FipaMessage, message)
+        dialogue_reference = fipa_msg.dialogue_reference
+
+        # new dialogue by other
+        if (
+            dialogue_reference[0] != ""
+            and dialogue_reference[1] == ""
```

Comment:
The dialogue reference object should expose a property `is_other_initiated` which provides this check. Same for `is_self_initiated`...then we hide this complexity and can alter it on the dialogue reference without affecting this implementation here.

Reply:
How to know, from `dialogue_reference` alone, if the dialogue is self_initiated vs other_initiated? Am I wrong in assuming that In both cases, the agent first creating the dialogue just fills in the first element of `dialogue_reference` and the receiver fills in the second element?

Reply:
I agree that If there is a way, then it should be done in the dialogue label class and hidden in the update method.